pkg/drivers/krunkit/krunkit.go

```diff
@@ -0,0 +1,562 @@
+//go:build darwin
+
+/*
+Copyright 2024 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package krunkit
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/docker/machine/libmachine/drivers"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/docker/machine/libmachine/mcnutils"
+	"github.com/docker/machine/libmachine/ssh"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/pkg/errors"
+
+	"k8s.io/klog/v2"
+	pkgdrivers "k8s.io/minikube/pkg/drivers"
+	"k8s.io/minikube/pkg/drivers/vmnet"
+	"k8s.io/minikube/pkg/minikube/exit"
+	"k8s.io/minikube/pkg/minikube/firewall"
+	"k8s.io/minikube/pkg/minikube/out"
+	"k8s.io/minikube/pkg/minikube/process"
+	"k8s.io/minikube/pkg/minikube/reason"
+	"k8s.io/minikube/pkg/minikube/style"
+)
+
+const (
+	driverName     = "krunkit"
+	isoFileName    = "boot2docker.iso"
+	pidFileName    = "krunkit.pid"
+	sockFileName   = "krunkit.sock"
+	logFileName    = "krunkit.log"
+	serialFileName = "serial.log"
+	logLevelInfo   = "3"
+	defaultSSHUser = "docker"
+)
+
+// Driver is the machine driver for krunkit (Virtualization.framework)
+type Driver struct {
+	*drivers.BaseDriver
+	*pkgdrivers.CommonDriver
+	Boot2DockerURL string
+	DiskSize       int
+	CPU            int
+	Memory         int
+	ExtraDisks     int
+	MACAddress     string
+	VmnetHelper    vmnet.Helper
+}
+
+func NewDriver(hostName, storePath string) drivers.Driver {
+	return &Driver{
+		BaseDriver: &drivers.BaseDriver{
+			SSHUser:     defaultSSHUser,
+			MachineName: hostName,
+			StorePath:   storePath,
+		},
+		CommonDriver: &pkgdrivers.CommonDriver{},
+	}
+}
+
+func (d *Driver) PreCreateCheck() error {
```

Comment:
same here ones bellow, all exported funcs need comment, I am surprised the linter didnt yell at us

Reply:
This was copied from vfkit, I guess this is missing in many places. I'll check how we document it in other places.